framework/integration/test/reconciliation/reconciliation_test.go

```diff
@@ -102,7 +102,7 @@ func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
 	//
 	// 		EnsureCreated: 3, EnsureDeleted: 0
 	//
-	time.Sleep(25 * time.Second)
+	time.Sleep(23 * time.Second)
```

Comment:
I think it can be improved by using backoff and checking CreateCount there. After that you can check it was ~22 seconds since test start and all other stuff.

Reply:
You are right, that's probably better. Going to do this.

Reply:
I changed this, but it still looks a bit ugly. I will think of something to structure this a bit better.